### PR TITLE
fix mmap IndexError

### PR DESCRIPTION
**Pull request by fengxuduke**
this bug will cause indexerror when offset+size>self.__mm.size()*self.__resizeFactor

### CODE REVIEW

File: pysyncobj/journal.py [hunk 1]
```diff
@@ -21,17 +21,6 @@ def deleteEntriesTo(self, entryTo):
 
     def __getitem__(self, item):
         raise NotImplementedError
-
```
Comment (bakwc) [inline pysyncobj/journal.py]: Why you removed this?

File: pysyncobj/journal.py [hunk 2]
```diff
@@ -80,9 +69,9 @@ def write(self, offset, values):
         currSize = self.__mm.size()
         if offset + size > self.__mm.size():
             try:
-                self.__mm.resize(int(self.__mm.size() * self.__resizeFactor))
```
Comment (bakwc) [inline pysyncobj/journal.py]: Could you please write unit-test for this case?

File: pysyncobj/syncobj.py [hunk 3]
```diff
@@ -1264,7 +1264,7 @@ def __tryLogCompaction(self):
             nodesCount = len(allNodes)
             selfIdx = allNodes.index(self.__selfNodeAddr)
             interval = self.__conf.logCompactionMinTime
-            periodStart = int(currTime) / interval * interval
```
Comment (bakwc) [inline pysyncobj/syncobj.py]: Could you please write unit-test for this case?

**Comment by bakwc** [comment]
Thanks for your PR. Could you please restore removed code and write unit-test for your fix?

**Comment by fengxuduke** [comment]
sure. will make a fix

**Pull request closed without merging**